### PR TITLE
Added Deft layout

### PR DESCRIPTION
**Pull request by Speiserm**
## Description
Added Deft to the Planck.

## Types of Changes
Added the Deft layout.

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* A lack of my layout

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/deft/keymap.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/*
+ * The Deft layout for the Planck Keyboard
+ * Version 1.0
+ *
+ * A keyboard layout deisgned for effeciency in programming and transferability between ortholinear boards.
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/deft/keymap.c]: Spelling corrections.

```suggestion
 * A keyboard layout designed for efficiency in programming and transferability between ortholinear boards.
```

File: keyboards/planck/keymaps/deft/keymap.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ * The Deft keyboard layout was designed by Michael Speiser. You can find other Deft layouts at my Github:
+ * https://github.com/Speiserm
+ *
+ * For questions, issues, or feedback, feel free to email me at Speiserm@pm.me.
+ *
+ *
+ * Flashing:
+ *
+ * Rev 5 or earlier:
+ * make planck/revX:mersault:dfu
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/deft/keymap.c]:
```suggestion
 * make planck/revX:deft:dfu
```

File: keyboards/planck/keymaps/deft/keymap.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ *
+ * Rev 6:
+ * make planck/rev6:mersault:dfu-util
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/deft/keymap.c]:
```suggestion
 * make planck/rev6:deft:dfu-util
```

File: keyboards/planck/keymaps/deft/readme.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# The Deft Planck Layout
+
+![Deft layout for the Planck keyboard](https://i.imgur.com/VtUkAyh.png)
+
+The Deft Planck layout is designed for effeciency in programming and transferability between ortholinear boards. It features a fairly standard default layer with properly placed arrow keys, a lower layer with a usable numpad and symbols, and a raise layer for numbers, f-keys, and music controls. The code is also cut down to only what you need. The result is a very usable Planck that can easily replace fullsize layouts.
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/deft/readme.md]: Another round of spell checking.

```suggestion
The Deft Planck layout is designed for efficiency in programming and transferability between ortholinear boards. It features a fairly standard default layer with properly placed arrow keys, a lower layer with a usable numpad and symbols, and a raise layer for numbers, f-keys, and music controls. The code is also cut down to only what you need. The result is a very usable Planck that can easily replace fullsize layouts.
```

File: keyboards/planck/keymaps/deft/readme.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+https://twitter.com/CatBucha
+
+https://www.instagram.com/buchathecat/
+
+
+## Flashing:
+| Rev 5 or earlier: | Rev 6: |
+|-|-|
+| make planck/revX:deft:dfu | make planck/rev6:deft:dfu-util |
```
Comment (fauxpark) [inline keyboards/planck/keymaps/deft/readme.md]:
```suggestion
| `make planck/revX:deft:dfu` | `make planck/rev6:deft:dfu-util` |
```

**Pull request merged**